docs/backend_architecture/upgrade.rst

```diff
@@ -18,7 +18,31 @@ and create a backup before doing so.
 
 .. note:: Always upgrade **as often as possible**. If you are responsible for
   deployments at different sites, you should consider a strategy for keeping
-  software and contents updated. 
+  software and contents updated.
+
+As well as database migrations, there are also sometimes additional fixes that
+are put into Kolibri in order to facilitate moving between versions. This may be
+for bug fixing or efficiency purposes. These are sometimes carried out outside of
+migrations in order to leverage the full Kolibri code base, which can be restricted
+inside the contexts of Django data migrations.
+
+In order to implement these upgrades, a decorator is available in ``kolibri.core.upgrade``,
+``version_upgrade``. An toy example is shown below.
+
+.. code-block:: python
+
+  import logging
+  from kolibri.core.upgrade import version_upgrade
+
+  logger = logging.getLogger(__name__)
+
+  @version_upgrade(old_version="<0.6.4", new_version=">=1.0.0")
```

Comment:
To clarify, are these "database versions" or Kolibri versions?

I think the rest of the content on this page is framed as user-facing instructions talking about Kolibri versions rather than developer docs talking about the `data_version`.

Reply:
These are Kolibri versions.

I found the inclusion of these instructions in our dev docs confusing, actually, because they did seem to be user facing, rather than dev facing.

Reply:
yes. I think they could use a re-work, because I'm not sure they're actually accurate.

Would you mind giving the the other stuff in there a quick look and remove if appropriate?

Reply:
I removed everything, and then added some bonus material about django schema migrations.